### PR TITLE
fix: properly sync all topics to cio

### PR DESCRIPTION
**Pull request by capJavert**
> When sending `cio_subscription_preferences` from `cio.identify` (for user) we have to include all existing topics in the object. If not CIO will take what it has and make everything else a default value (that is set in cio topics area) and for stuff like digest and follow it will become "subscribed" again.

Above was hotfixed in https://github.com/dailydotdev/daily-api/pull/2314

In addition to above, this PR also sync other subs in the same way. 

In addition it also syncs digest after subscribe/unsubscribe mutations through `identifyUserPersonalizedDigest` util.

More discussion in [thread](https://dailydotdev.slack.com/archives/C01L0QXQJNB/p1728038592931229)

### CODE REVIEW

**Review by rebelchris — APPROVED** [review]
Nice fix!

File: src/cio.ts [hunk 1]
```diff
@@ -81,18 +87,34 @@ export async function identifyUserStreak(
...
+  con,
+  log,
+  cio,
+  user,
+}: {
+  con: DataSource;
+  log: FastifyBaseLogger;
+  cio: TrackClient;
+  user: ChangeObject<User>;
+}): Promise<void> {
```
Comment (capJavert) [inline src/cio.ts:100]: Refactored to props syntax because I had to add `con`, we prefer it either way.

Comment (rebelchris) [reply src/cio.ts:100]: Nice yeah we should just use object notation by default 🤦

File: src/schema/users.ts [hunk 2]
```diff
@@ -1790,7 +1790,14 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
...
 
-      await resubscribeUser(cio, ctx.userId);
+      await Promise.all([
+        resubscribeUser(cio, ctx.userId),
+        identifyUserPersonalizedDigest({
+          userId: ctx.userId,
+          cio,
+          subscribed: true,
+        }),
+      ]);
```
Comment (capJavert) [inline src/schema/users.ts:1800]: Subscribes to digest in CIO

File: src/schema/users.ts [hunk 3]
```diff
@@ -1810,6 +1817,12 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
         });
       }
 
+      await identifyUserPersonalizedDigest({
+        userId: ctx.userId,
+        cio,
+        subscribed: false,
+      });
```
Comment (capJavert) [inline src/schema/users.ts:1824]: Unsubscribes from digest in CIO

File: src/schema/users.ts [hunk 4]
```diff
@@ -1790,7 +1790,14 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
         flags,
       });
 
-      await resubscribeUser(cio, ctx.userId);
+      await Promise.all([
+        resubscribeUser(cio, ctx.userId),
+        identifyUserPersonalizedDigest({
```
Comment (rebelchris) [inline src/schema/users.ts:1795]: Doesn't `identifyUser` already do all of it?

**Review by rebelchris — APPROVED** [review]
Overall good, just wondering if we should just only have the 1 generic function that does all of them at once?

File: src/schema/users.ts (hunk 4 above)
Comment (capJavert) [reply src/schema/users.ts:1795]: it does the digest part but I would have to fetch user data here to be able to use it so to me it made more sense to create a separate helper to also be able to jjst set digest part of subscription.

Comment (rebelchris) [reply src/schema/users.ts:1795]: Ah ok let's start lean then :)

File: src/workers/userUpdatedCio.ts [hunk 5]
```diff
@@ -24,7 +24,12 @@ const worker: TypedWorker<'user-updated'> = {
       await resubscribeUser(cio, user.id);
     }
 
-    await identifyUser(log, cio, user);
+    await identifyUser({
+      con,
+      log,
```
Comment (omBratteng) [inline src/workers/userUpdatedCio.ts:29]: we can remove log, and just import logger. So we don't have to keep passing it around

Comment (capJavert) [reply src/workers/userUpdatedCio.ts:29]: yeah, I used it in new function, will also add here while I am refactoring it either way

Comment (capJavert) [reply src/workers/userUpdatedCio.ts]: Done in 9fe14dff7e5a4186290b734209f74fe4200368a5

**Review by omBratteng — APPROVED** [review]

**Review by idoshamun — APPROVED** [review]

**Pull request merged**